### PR TITLE
[Data] Update code owners of Ray Data

### PR DESCRIPTION
**Pull request by c21**
## Why are these changes needed?

As title, to reflect the latest group actively working on Ray Data module.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -75,10 +75,8 @@
 # ==== Libraries and frameworks ====
 
 # Ray data.
-/python/ray/data/ @ericl @scv119 @clarkzinzow @jjyao @jianoaix @c21
-/python/ray/data/preprocessors/ @clarkzinzow @jiaodong @Yard1 @bveeramani @matthewdeng @amogkam
```
Comment (c21) [inline .github/CODEOWNERS:79]: I don't think we need a special group for `preprocessors` sub-directory, so I removed them here.

**Comment by jjyao** [comment]
You should remove me as well :)

**Comment by c21** [comment]
> You should remove me as well :)

All right, done. Thanks @jjyao. We will miss you.

**Review by scottjlee — APPROVED** [review]

**Review by ericl — APPROVED** [review]

**Comment by c21** [comment]
Merged to master.

**Pull request merged**